src/models/frame.ts

```diff
@@ -39,57 +70,55 @@ export class FrameNavigation {
  */
 export class FrameStore {
   private frames: { [frameId: string]: FrameNavigation };
-  private loadResolver: () => void;
 
   constructor() {
     this.frames = {};
   }
-  start(frameId: string, waitForLoad: boolean = false): Promise<PageNavigateResult> {
-    const nav = new FrameNavigation();
+  start(frameId: string, waitForLoad: boolean): Promise<Network.ResponseReceivedParameters[]> {
+    const nav = new FrameNavigation(waitForLoad);
     this.frames[frameId] = nav;
     const navPromise = nav.getPromise();
-    return waitForLoad ?
-      Promise.all([navPromise, new Promise((r) => { this.loadResolver = r; })]).then(([n]) => n) :
-      navPromise;
+    return navPromise;
   }
   onNetworkResponse(res: Network.ResponseReceivedParameters) {
     if (res.frameId) {
       const nav = this.frames[res.frameId];
-      // TODO: Figure out why sometimes we recieve a bogus requestId that is a
-      // decimal number represented as a string
-      if (nav && res.requestId.indexOf('.') < 0) {
+      if (nav) {
         nav.onNetworkResponse(res);
       }
     } else {
       throw new Error('Received network response without frameId');
     }
   }
+  onRequestWillBeSent(req: Network.RequestWillBeSentParameters) {
+    if (req.frameId) {
+      const nav = this.frames[req.frameId];
+      if (nav) {
+        nav.onRequestWillBeSent(req);
+      }
+    } else {
+      throw new Error('Received network request will be sent without frameId');
```

Comment:
This error message makes it sound like the issue is a recoverable warning, when we are doing a hard throw in reality

Reply:
I've reworded it to 'Received "request will be sent" event without frameId'.

It is already classified as an error by the use of Error(). I think that should be enough? I thought it redundant to say Error("Error: Received....